fw/connection.c

```diff
@@ -174,7 +174,9 @@ tfw_connection_recv(TfwConn *conn, struct sk_buff *skb)
 				r = tfw_ws_msg_process(conn, skb);
 			else
 				r = tfw_http_msg_process(conn, skb, &splitted);
-			if (r == T_DROP && splitted) {
+
+			if (splitted && (r == T_DROP || r == T_BLOCK_WITH_RST
```

Comment:
r can be also -ERRNO  here, I think that we should not check `r` here, but `splitted` should be zero if r == T_OK or r == T_POSTPONED. Please check it.